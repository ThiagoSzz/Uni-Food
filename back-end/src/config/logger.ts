import winston from "winston";
import colors from "colors/safe";

interface LogColors {
  [key: string]: (text: string) => string;
}

const logColors: LogColors = {
  info: colors.blue,
  warn: colors.yellow,
  error: colors.red,
};

export const logger = winston.createLogger({
  format: winston.format.combine(
    winston.format.timestamp(),
    winston.format.printf(({ level, message, timestamp }) => {
      const coloredLevel = logColors[level.toLowerCase()](level.toUpperCase());
      return `[${timestamp}] ${coloredLevel}: ${message}`;
    })
  ),
  transports: [new winston.transports.Console()],
});
