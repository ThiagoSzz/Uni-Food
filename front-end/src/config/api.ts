import axios from 'axios';

const api = axios.create({
  baseURL: process.env.REACT_APP_API_URL || 'http://localhost:8080/api'
});

// Request interceptor to add authentication token
api.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('auth_token');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

// Response interceptor to handle authentication errors
api.interceptors.response.use(
  (response) => {
    return response;
  },
  (error) => {
    if (error.response?.status === 401 || error.response?.status === 403) {
      // Token is invalid or expired, clear authentication data
      localStorage.removeItem('auth_token');
      localStorage.removeItem('auth_user');

      if (window.location.pathname !== '/') {
        window.location.href = '/';
      }
    }
    return Promise.reject(error);
  }
);

export default api;
