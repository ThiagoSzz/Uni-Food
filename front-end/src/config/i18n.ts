import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

// Minimal loader for properties-like file already in public
async function loadProperties(): Promise<Record<string, string>> {
  try {
    const res = await fetch('/i18n.properties');
    const text = await res.text();
    const lines = text.split(/\r?\n/);
    const map: Record<string, string> = {};
    for (const line of lines) {
      const trimmed = line.trim();
      if (!trimmed || trimmed.startsWith('#')) continue;
      const idx = trimmed.indexOf('=');
      if (idx === -1) continue;
      const key = trimmed.substring(0, idx).trim();
      const val = trimmed.substring(idx + 1).trim();
      map[key] = val;
    }
    return map;
  } catch (e) {
    console.error('Failed to load i18n.properties', e);
    return {};
  }
}

export async function initI18n() {
  const resources = { pt: { translation: await loadProperties() } };

  await i18n.use(initReactI18next).init({
    resources,
    lng: 'pt',
    fallbackLng: 'pt',
    interpolation: { escapeValue: false },
    react: { useSuspense: false }
  });
}

export default i18n;
