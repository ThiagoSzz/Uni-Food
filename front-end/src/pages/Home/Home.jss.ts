/* eslint-disable @typescript-eslint/no-unused-vars */
import { createUseStyles } from 'react-jss';

type JSSStyles = { [keys: string]: React.CSSProperties | JSSStyles };

const styles: JSSStyles = {
  boxesContainer: {
    gap: '20px',
    justifyContent: 'center',
    marginTop: '20px'
  },
  textContainer: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    marginTop: '20px'
  },
  centeredContainer: {
    width: '1354px',
    gap: '10px',
    flexWrap: 'wrap',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    overflowY: 'auto',
    '@media (min-width: 1024px) and (max-width: 1439px)': {
      width: '1010px'
    }
  },
  sectionText: {
    alignSelf: 'center',
    marginRight: '10px',
    marginBottom: '20px'
  },
  busyIndicator: {
    position: 'absolute',
    alignSelf: 'center',
    left: '50%'
  },
  averageReviewsContainer: {
    gap: '20.5px',
    width: '1360px',
    height: '150px',
    padding: '2px',
    flexWrap: 'wrap',
    alignSelf: 'center',
    justifyContent: 'flex-start',
    overflowY: 'auto',
    '@media (min-width: 1024px) and (max-width: 1439px)': {
      width: '1014px'
    }
  },
  reviewsContainer: {
    gap: '20.5px',
    width: '1360px',
    height: '730px',
    padding: '2px',
    flexWrap: 'wrap',
    alignSelf: 'center',
    justifyContent: 'flex-start',
    overflowY: 'auto',
    marginBottom: '40px',
    '@media (min-width: 1024px) and (max-width: 1439px)': {
      width: '1014px'
    }
  }
};

export const useStyles = createUseStyles(styles, {
  name: 'Home'
});
