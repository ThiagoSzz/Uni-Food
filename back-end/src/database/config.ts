import { neon } from "@neondatabase/serverless";
import { config } from "dotenv";
import { logger } from "../config/logger";

config();

const sql = neon(process.env.DATABASE_URL!);

export const sqlOperation = async (
  operation: string,
  values: any[],
  key: string
) => {
  logger.info(`SQL Operation is running${key ? ": " + `${key}` : ""}`);

  const result = await sql(operation, values);

  return result;
};
