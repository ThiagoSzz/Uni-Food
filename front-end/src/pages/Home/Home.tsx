/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import {
  BusyIndicator,
  FlexBox,
  FlexBoxDirection,
  Text,
  Title,
  TitleLevel,
  MessageStrip,
  MessageStripDesign
} from '@ui5/webcomponents-react';

import { useStyles } from './Home.jss';
import '@ui5/webcomponents-icons/dist/AllIcons';
import { CustomShellBar } from '../../components/ShellBar/CustomShellBar/CustomShellBar';
import { ReviewCard } from '../../components/ReviewCard/ReviewCard';
import { AverageReviewsCard } from '../../components/AverageReviewsCard/AverageReviewsCard';
import { FilterDialog } from '../../components/FiltersDialog/FiltersDialog';
import { CreateReviewInfoBox } from '../../components/CreateReviewInfoBox/CreateReviewInfoBox';
import { SearchReviewInfoBox } from '../../components/SearchReviewInfoBox/SearchReviewInfoBox';
import { ReviewsSearchBar } from '../../components/ReviewsSearchBar/ReviewsSearchBar';
import useNewReviewStore from '../../stores/useNewReviewStore';
import useAverageReviewsStore from '../../stores/useAverageReviewsStore';
import useReviewsStore from '../../stores/useReviewsStore';
import { useAverageReviews } from '../../hooks/useAverageReviews';
import { useGetReviewsMutation } from '../../hooks/queries/useGetReviews';
import { getReviewsList } from '../../fixtures/ReviewsFixture';

const USE_BACKEND_REVIEWS = true;
const NUM_DISPLAYED_REVIEWS = 40;

export const Home: React.FC = () => {
  const classes = useStyles();

  const [clearValidationErrors, isReviewCreated, setIsReviewCreated] = useNewReviewStore(
    (value) => [value.clearValidationErrors, value.isReviewCreated, value.setIsReviewCreated]
  );
  const [
    reviews,
    setReviews,
    filteredReviews,
    setFilteredReviews,
    shouldShowNoFilteredReviewsMessage,
    searchQuery
  ] = useReviewsStore((value) => [
    value.reviews,
    value.setReviews,
    value.filteredReviews,
    value.setFilteredReviews,
    value.shouldShowNoFilteredReviewsMessage,
    value.searchQuery
  ]);
  const [setAverageReviews, filteredAverageReviews, setFilteredAverageReviews] =
    useAverageReviewsStore((value) => [
      value.setAverageReviews,
      value.filteredAverageReviews,
      value.setFilteredAverageReviews
    ]);

  const [isLoadingReviews, setIsLoadingReviews] = useState<boolean>(true);
  const [isLoadingAverageReviews, setIsLoadingAverageReviews] = useState<boolean>(true);
  const [feedbackMessageStrip, setFeedbackMessageStrip] = useState<{
    show: boolean;
    message?: string;
    value?: MessageStripDesign;
  }>({
    show: false
  });

  const { getUniversityRuStandings, groupReviewsByRuAndUniversity } = useAverageReviews();

  const reviewsMutation = useGetReviewsMutation();

  const fetchReviews = async () => {
    await reviewsMutation.mutateAsync().then((result) => {
      const reviews = result.data;

      setReviews(reviews);
      setFilteredReviews(reviews);
    });
  };

  useEffect(() => {
    if (shouldShowNoFilteredReviewsMessage) {
      setFeedbackMessageStrip({
        show: true,
        message: 'Nenhuma avaliação foi filtrada com sua pesquisa.',
        value: MessageStripDesign.Information
      });

      const timeoutId = setTimeout(() => {
        setFeedbackMessageStrip({ show: false });
      }, 5000);

      return () => {
        clearTimeout(timeoutId);
      };
    }
  }, [shouldShowNoFilteredReviewsMessage]);

  useEffect(() => {
    setFeedbackMessageStrip({ show: false });
  }, [searchQuery]);

  useEffect(() => {
    if (isReviewCreated) {
      setFeedbackMessageStrip({
        show: true,
        message: 'Sua avaliação foi criada com sucesso!',
        value: MessageStripDesign.Positive
      });

      setIsReviewCreated(false);
    }

    setIsLoadingReviews(true);
    setIsLoadingAverageReviews(true);
    clearValidationErrors();

    if (USE_BACKEND_REVIEWS) {
      fetchReviews();
    } else {
      const reviewsFixture = getReviewsList();

      setReviews(reviewsFixture);
      setFilteredReviews(reviewsFixture);
    }
  }, []);

  useEffect(() => {
    if (reviews.length > 0) {
      const bestAndWorstRusByUniversity = getUniversityRuStandings(reviews);
      const reviewsGroupedByRuAndUniversity = groupReviewsByRuAndUniversity(
        reviews,
        bestAndWorstRusByUniversity
      );

      setAverageReviews(reviewsGroupedByRuAndUniversity);
      setFilteredAverageReviews(reviewsGroupedByRuAndUniversity);
    }
  }, [reviews]);

  useEffect(() => {
    if (filteredAverageReviews.length > 0) {
      setIsLoadingReviews(false);
      setIsLoadingAverageReviews(false);
    }
  }, [filteredAverageReviews]);

  useEffect(() => {
    if (feedbackMessageStrip.show) {
      const timeoutId = setTimeout(() => {
        setFeedbackMessageStrip({ show: false });
      }, 5000);

      return () => {
        clearTimeout(timeoutId);
      };
    }
  }, [feedbackMessageStrip.show]);

  return (
    <FlexBox direction={FlexBoxDirection.Column}>
      <CustomShellBar searchDisabled />

      <FlexBox direction={FlexBoxDirection.Column}>
        <FlexBox direction={FlexBoxDirection.Row} className={classes.boxesContainer}>
          <CreateReviewInfoBox />
          <SearchReviewInfoBox />
        </FlexBox>
        {feedbackMessageStrip.show ? (
          <FlexBox className={classes.messageStripContainer}>
            <MessageStrip
              className={classes.messageStrip}
              onClose={() => setFeedbackMessageStrip({ show: false })}
              design={feedbackMessageStrip.value}
            >
              <Text style={{ fontSize: '15px' }}>{feedbackMessageStrip.message}</Text>
            </MessageStrip>
          </FlexBox>
        ) : (
          <></>
        )}
        <FlexBox
          className={classes.boxesContainer}
          style={{ marginTop: '10px', marginBottom: '0px' }}
        >
          <ReviewsSearchBar />
          <FilterDialog />
        </FlexBox>
        <FlexBox className={classes.textContainer}>
          <FlexBox className={classes.centeredContainer}>
            <Title className={classes.sectionText} level={TitleLevel.H4}>
              Médias por Restaurante Universitário
            </Title>
            <Text className={classes.sectionText}>
              ({!isLoadingAverageReviews ? filteredAverageReviews.length : '??'} RUs)
            </Text>
          </FlexBox>
        </FlexBox>
        <FlexBox className={classes.averageReviewsContainer}>
          <BusyIndicator
            delay={0}
            active={isLoadingAverageReviews}
            className={classes.busyIndicator}
          />
          {!isLoadingAverageReviews &&
            filteredAverageReviews.map((averageReview, index) => {
              return <AverageReviewsCard key={index} averageReview={averageReview} />;
            })}
        </FlexBox>
        <FlexBox className={classes.textContainer}>
          <FlexBox className={classes.centeredContainer}>
            <Title className={classes.sectionText} level={TitleLevel.H4}>
              Avaliações por Refeição
            </Title>
            <Text className={classes.sectionText}>
              ({!isLoadingReviews ? filteredReviews.length : '??'} avaliações)
            </Text>
          </FlexBox>
        </FlexBox>
        <FlexBox className={classes.reviewsContainer}>
          <BusyIndicator
            delay={0}
            active={isLoadingReviews}
            className={classes.busyIndicator}
            style={{ marginBottom: '300px' }}
          />
          {!isLoadingReviews &&
            filteredReviews.slice(0, NUM_DISPLAYED_REVIEWS).map((review, index) => {
              return <ReviewCard key={index} review={review} />;
            })}
        </FlexBox>
      </FlexBox>
    </FlexBox>
  );
};
