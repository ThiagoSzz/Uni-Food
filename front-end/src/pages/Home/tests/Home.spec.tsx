import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { QueryClientProvider } from 'react-query';
import { queryClient } from '../../../config/client';
import { ThemeProvider } from '@ui5/webcomponents-react';
import { expect } from '@jest/globals';
import { Home } from '../Home';

describe('Home page', () => {
  const setup = () => {
    return render(
      <QueryClientProvider client={queryClient}>
        <MemoryRouter>
          <ThemeProvider>
            <Home />
          </ThemeProvider>
        </MemoryRouter>
      </QueryClientProvider>
    );
  };

  it('Should match the snapshot', () => {
    const view = setup();
    expect(view.container).toMatchSnapshot();
  });

  it('Should load the search bar with the filters and switches', () => {});

  it('Should load the list of reviews', () => {});

  it('Should show the amount of reviews', () => {});

  it('Should load the list of average reviews', () => {});

  it('Should show the amount of average reviews', () => {});
});
