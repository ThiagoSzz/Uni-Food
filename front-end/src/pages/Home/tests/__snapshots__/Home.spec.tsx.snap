// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Should match the snapshot 1`] = `
<div>
  <div
    class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
  >
    <ui5-shellbar
      data-sap-ui-fastnavgroup="true"
      primary-title="UniFood"
      ui5-shellbar=""
    >
      <img
        alt="UniFood Logo"
        slot="logo"
        src="https://static-00.iconduck.com/assets.00/fork-and-knife-with-plate-emoji-2048x2048-4e58vsav.png"
      />
      <ui5-avatar
        color-scheme="Accent6"
        initials="TH"
        slot="profile"
        ui5-avatar=""
      />
      <ui5-shellbar-item
        icon="visits"
        text="Entrar"
        ui5-shellbar-item=""
      />
      <ui5-shellbar-item
        icon="add-employee"
        text="Cadastrar-se"
        ui5-shellbar-item=""
      />
    </ui5-shellbar>
    <div
      class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionRow-0-2-44 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-firstSection-0-2-1"
    >
      <div
        class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-mainContainer-0-2-2"
      >
        <span
          class="Text-text-0-2-49 Home-firstSectionMainText-0-2-4"
        >
          Compartilhe suas experiências gastronômicas em restaurantes universitários!
        </span>
        <span
          class="Text-text-0-2-49 Home-firstSectionSecondaryText-0-2-5"
        >
          Sua opinião importa! Junte-se a nós e ajude a transformar a experiência gastronômica estudantil nas universidades brasileiras.
        </span>
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionRow-0-2-44 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-buttonsFlexBox-0-2-6"
        >
          <ui5-button
            class="Home-newReviewButton-0-2-7"
            design="Emphasized"
            type="Button"
            ui5-button=""
          >
            Deixar Avaliação
          </ui5-button>
          <ui5-button
            class="Home-searchReviewsButton-0-2-8"
            design="Default"
            type="Button"
            ui5-button=""
          >
            Pesquisar Avaliações
          </ui5-button>
        </div>
      </div>
      <div
        class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionRow-0-2-44 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-secondaryContainer-0-2-3"
      >
        <ui5-card
          class="Home-reviewCard-0-2-12"
          ui5-card=""
        >
          <ui5-card-header
            slot="header"
            subtitle-text="Porto Alegre, Rio Grande do Sul"
            title-text="RU1 - UFRGS"
            ui5-card-header=""
          >
            <ui5-icon
              design="Default"
              name="official-service"
              slot="avatar"
              ui5-icon=""
            />
          </ui5-card-header>
          <ui5-list
            accessible-role="list"
            busy-delay="1000"
            data-sap-ui-fastnavgroup="true"
            growing="None"
            mode="None"
            separators="None"
            ui5-list=""
          >
            <ui5-li
              additional-text-state="None"
              type="Inactive"
              ui5-li=""
            >
              <ui5-rating-indicator
                max="5"
                readonly="true"
                ui5-rating-indicator=""
                value="4"
              />
            </ui5-li>
            <ui5-li
              additional-text-state="None"
              style="height: auto; margin-top: 20px;"
              type="Inactive"
              ui5-li=""
            >
              <h4>
                Comentários
              </h4>
              <p
                class="Home-reviewCardComments-0-2-13"
              >
                Minha experiência no Restaurante Universitário foi positiva em vários aspectos. A agilidade no atendimento, com filas curtas, facilitou bastante o processo de refeição. No entanto, a ausência de guardanapos foi um ponto negativo que não pude ignorar. A diversidade de pratos disponíveis é uma vantagem, embora tenha sentido falta de opções vegetarianas para atender diferentes preferências alimentares.
              </p>
            </ui5-li>
            <ui5-li
              additional-text-state="None"
              style="height: auto; margin-top: 20px;"
              type="Inactive"
              ui5-li=""
            >
              <h4>
                Tags
              </h4>
              <div
                class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionRow-0-2-44 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-badgesList-0-2-14"
              >
                <ui5-badge
                  class="Home-badge-0-2-15"
                  color-scheme="8"
                  ui5-badge=""
                >
                  Fila curta
                </ui5-badge>
                <ui5-badge
                  class="Home-badge-0-2-15"
                  color-scheme="1"
                  ui5-badge=""
                >
                  Sem guardanapos
                </ui5-badge>
                <ui5-badge
                  class="Home-badge-0-2-15"
                  color-scheme="2"
                  ui5-badge=""
                >
                  Sem opção vegetariana
                </ui5-badge>
                <ui5-badge
                  class="Home-badge-0-2-15"
                  color-scheme="8"
                  ui5-badge=""
                >
                  Boa variedade
                </ui5-badge>
                <ui5-badge
                  class="Home-badge-0-2-15"
                  color-scheme="8"
                  ui5-badge=""
                >
                  Limpeza
                </ui5-badge>
              </div>
            </ui5-li>
          </ui5-list>
        </ui5-card>
      </div>
    </div>
    <div
      class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-secondSection-0-2-9"
    >
      <span
        class="Text-text-0-2-49 Home-secondSectionMainText-0-2-10"
      >
        Veja as avaliações dos RUs mais frequentados!
      </span>
      <span
        class="Text-text-0-2-49 Home-secondSectionSecondaryText-0-2-11"
      >
        Veja a média das avaliações deixadas por outros estudantes.
      </span>
    </div>
    <div
      class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-thirdSection-0-2-16"
    >
      <span
        class="Text-text-0-2-49 Home-secondSectionMainText-0-2-10"
      >
        Leia avaliações reais de estudantes como você
      </span>
      <span
        class="Text-text-0-2-49 Home-secondSectionSecondaryText-0-2-11"
      >
        Descubra as experiências mais recentes dos estudantes nos RUs.
      </span>
    </div>
    <div
      class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-fourthSection-0-2-19"
    >
      <div
        class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-fourthSectionMainContainer-0-2-20"
      >
        <ui5-badge
          color-scheme="8"
          style="padding: 5px;"
          ui5-badge=""
        >
          FAQ
        </ui5-badge>
        <span
          class="Text-text-0-2-49 Home-fourthSectionMainText-0-2-21"
        >
          Perguntas comuns
        </span>
      </div>
      <div
        class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        style="padding: 100px; gap: 50px;"
      >
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        >
          <span
            class="Text-text-0-2-49 Home-questionText-0-2-23"
          >
            O que devo considerar em minha avaliação?
          </span>
          <span
            class="Text-text-0-2-49 Home-answerText-0-2-24"
          >
            Ao avaliar o restaurante universitário, você pode considerar diversos aspectos, tais como a qualidade da comida, o atendimento, a limpeza do ambiente e a variedade de opções oferecidas. 
            <br />
             
            <br />
            É importante que você leve em conta a realidade dos RUs ao fazer comparações, evitando considerar restaurantes que podem ter preços muito mais altos. Afinal, o valor pago no RU é significativamente mais acessível e a proposta do local é atender às necessidades dos estudantes.
          </span>
        </div>
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        >
          <span
            class="Text-text-0-2-49 Home-questionText-0-2-23"
          >
            Meu nome ficará exposto na avaliação?
          </span>
          <span
            class="Text-text-0-2-49 Home-answerText-0-2-24"
          >
            Não, todas as avaliações são anônimas. Seu nome não será exibido e sua identidade será mantida em sigilo.
          </span>
        </div>
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        >
          <span
            class="Text-text-0-2-49 Home-questionText-0-2-23"
          >
            Qual é o objetivo da plataforma?
          </span>
          <span
            class="Text-text-0-2-49 Home-answerText-0-2-24"
          >
            O objetivo da plataforma é proporcionar um espaço para que estudantes possam compartilhar suas experiências nos restaurantes universitários, permitindo que outros alunos conheçam as avaliações e façam escolhas informadas sobre onde comer. 
            <br />
            <br />
             Além disso, acreditamos que os orgãos gerenciadores dos RUs avaliados pelos usuários em nossa plataforma podem contribuir para a evolução dos restaurantes de um modo geral. Dessa forma, as avaliações podem servir como uma métrica de qualidade destes estabelecimentos.
          </span>
        </div>
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        >
          <span
            class="Text-text-0-2-49 Home-questionText-0-2-23"
          >
            Durante quanto tempo uma avaliação fica disponível?
          </span>
          <span
            class="Text-text-0-2-49 Home-answerText-0-2-24"
          >
            Uma avaliação fica disponível por até um mês após ser enviada. Nós optamos por manter as avaliações temporárias porque acreditamos que os RUs têm a capacidade de evoluir e se adaptar às necessidades dos estudantes ao longo do tempo.
          </span>
        </div>
        <div
          class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
        >
          <span
            class="Text-text-0-2-49 Home-questionText-0-2-23"
          >
            Sou estudante e gostaria de obter acesso às avaliações, é possível?
          </span>
          <span
            class="Text-text-0-2-49 Home-answerText-0-2-24"
          >
            Sim, é possível obter acesso às avaliações em nossa plataforma. Se você é estudante e deseja ter acesso às avaliações, por favor, entre em contato conosco pelo e-mail
             
            <u>
              unifood.tech@gmail.com
            </u>
            . Vamos avaliar sua solicitação e fornecer os detalhes sobre as condições de utilização dos dados de avaliações em nossa plataforma. 
            <br />
            <br />
             Lembramos que é fundamental respeitar a Lei Geral de Proteção de Dados (LGPD) e as políticas de privacidade em vigor. Portanto, as informações fornecidas a você estarão de acordo com as regulamentações e garantirão a privacidade e anonimato dos avaliadores.
          </span>
        </div>
      </div>
    </div>
    <div
      class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionRow-0-2-44 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46 Home-footer-0-2-25"
    >
      <span
        class="Text-text-0-2-49 Home-footerText-0-2-26"
      >
        A missão da plataforma UniFood é tornar a experiência gastronômica universitária ainda mais especial e prática. Nossa proposta é simples: fornecer um espaço onde você possa avaliar os restaurantes universitários (RUs) com facilidade e compartilhar suas experiências. Suas avaliações podem contribuir para melhorar a experiência de todos que utilizam os mesmos RUs que você!
      </span>
      <div
        class="FlexBox-flexBox-0-2-29 FlexBox-flexBoxDirectionColumn-0-2-42 FlexBox-justifyContentStart-0-2-32 FlexBox-alignItemsStretch-0-2-40 FlexBox-flexWrapNoWrap-0-2-46"
      >
        <span
          class="Text-text-0-2-49 Home-footerContactHeader-0-2-27"
        >
          Contato: 
        </span>
        <span
          class="Text-text-0-2-49 Home-footerContact-0-2-28"
        >
          unifood.tech@gmail.com
        </span>
      </div>
    </div>
  </div>
</div>
`;
